src/actions/badgesActions.js

```diff
@@ -79,3 +80,32 @@ export const selfAwardBadgeAction = (badgeType: string) => {
     dispatch(offlineApiCall('selfAwardBadge', walletId, badgeType));
   };
 };
+
+export const fetchBadgeAwardHistoryAction = () => {
+  return async (dispatch: Dispatch, getState: GetState, api: SDKWrapper) => {
+    const {
+      user: { data: { walletId } },
+    } = getState();
+    const badgeAwardEvents = await api.fetchNotifications(walletId, BADGE_REWARD_EVENT) || [];
```

Comment:
I would force `fetchNotifications()` method to always return an array instead of writing `|| []`